src/deviceinfo.cpp

```diff
@@ -355,4 +358,47 @@ QString DeviceInfo::wlanMacAddress()
     return d->wlanMacAddress();
 }
 
+static QString normalizeUid(const QString &uid)
+{
+    // Normalize by stripping colons, dashes and making it lowercase
+    return uid.trimmed().replace(":", "").replace("-", "").toLower();
```

Comment:
Would require mid-string spaces for it to matter (which probably is never going to happen), but I'd still consider moving trimming to be the last op. Say, to cover likes of "foo - : -" -> "foo   " vs "foo"

Reply:
Sure.